### PR TITLE
Disable global install and verify folder structure

### PR DESCRIPTION
**Pull request by nason**
Re #379 

I didn't see how commands like `new` would work from a local install, so I kept that behavior and added some failure paths for non ignored commands in the case of:

- No local install
- Missing required files

Using a global install:
screen shot 2016-11-29 at 1 37 50 am

In the same directory after installing (linking) locally:
screen shot 2016-11-29 at 1 41 20 am

Let me know if you have any concerns and how you'd like to proceed!

### CODE REVIEW

**Comment by KyleAMathews** [comment]
Looking really good! I'll probably have time to merge tomorrow. Looks like there's some linting problems to fix first.

**Comment by nason** [comment]
Thanks @KyleAMathews! I'll look into the linting errors tonight

File: bin/gatsby.js [hunk 1]
```diff
@@ -30,33 +31,53 @@ var loadGatsby = function (path) {
...
+/*
+  These commands can:
+  - Use a global install as a fallback if no local install is present
+  - Bypass the required file check
+ */
+const commandsToIgnore = [undefined, 'new', '--help', '-h', '--version', '-V']
+const isIgnored = commandsToIgnore.includes(process.argv[2])
+
+const requiredFiles = [
+  'html.js',
```
Comment (nason) [inline bin/gatsby.js]: I realized that this approach requires a `.js` extension and isn't really flexible if you want to use `.jsx` for example. Do you have any thoughts on that?

Comment (KyleAMathews) [inline bin/gatsby.js]: Perhaps if we could modify the required file checker to support glob syntax then we could just do `'html.js*'`

Comment (KyleAMathews) [inline bin/gatsby.js]: @nason what do you think about this approach? Cool if you don't have time to do this last tweak — I can push this PR over the edge tomorrow or the next day when I do a new release.

Comment (KyleAMathews) [inline bin/gatsby.js]: @nason would you still like to push this PR to completion? Or return it the community for someone else to take over?

**Comment by KyleAMathews** [comment]
This is pretty ancient. We're probably going to switch to a `gatsby-cli` package for v1 anyways.

**Pull request closed without merging**